cin_validator/rules/cin2022_23/rule_2991Q.py

```diff
@@ -34,11 +34,12 @@ def validate(
     df_ass.reset_index(inplace=True)
     df_47.reset_index(inplace=True)
 
+    #  If <Section47> module is present then <Assessment> module should be present.
     merged_df = df_47.merge(
         df_ass,
         on=[LAchildID, CINdetailsID],
         suffixes=["_47", "_ass"],
-        how="outer",
+        how="left",
```

Comment:
It's okay to leave it as `outer` since the next line filters out the rows where the i`ndicator=="left_only"`. Hence, only the rows found in df_47 that are not found in df_ass will be left.
<img width="254" alt="image" src="https://user-images.githubusercontent.com/42181805/225345156-c972f86b-c504-4f8a-a55f-b8cf1a5090db.png">

Reply:
Isn't it more efficient to remove the unnnecessary data at the earlier stage (merge) rather than pulling through irrelevant assessments just to discard them immediately afterwards?

Reply:
You're right. Though the change in time/memory used will be quite small since we're working with small datasets, simply doing the left join is more efficient. 
Either way works. 
- outer join and filter by indicator or
- left join and remove indicator and filter line(s).
Whichever Will chooses is fine by me.

Reply:
My latest merge has gone back to outer so, rather than changing it again, lets leave it